### ISSUE TITLE
Enhance wine-staging scripts to support stacked git in addition to bare git

### ISSUE THREAD

**Issue by titan73** [wine-compholio/wine-staging#239]
Currently wine-staging scrips can commit the whole patchset or a subset directly to the git repo. To remove them 'git reset --hard' need to be done.

I would like something similar but with stacked git. The script needs to use stacked git command as it doesn't like someone else changing the repo.

So basically to reset all patches:
1) parse patch with 'stg serie', pop them if they are unapplied with 'stg pop' and delete them with 'stg delete'

So basically to apply patch:
1) make sure there is no wine-staging patch (applied or not)
    => do a reset (see below) or error message
2) applied requested patches
    => probably with 'stg import' + 'stg applied'
3) other patches will go in stacked git in unapplied state
    => probably with 'stg import' only

What do you think? I'd like have this to manage both own & wine-staging patches.

**Comment by titan73** [comment]
With this patch http://ix.io/fzV I got wine-staging patches to apply.
To remove I used:
stg serie | sed -e 's/^..//' | grep "^wine-staging" | xargs stg delete

**Comment by titan73** [comment]
The subject needs to be unique but sometime it's not due to the the new line.

**Comment by slackner** [comment]
Basic stacked git support was added in d4fa1d1816fe98c2dfe01d8b803de3ee1244069f.

To apply the patches you can use for example:
```
# All patches
/path/to/patches/patchinstall.sh DESTDIR="$(pwd)" --backend=stg --all

# Only a specific patchset
/path/to/patches/patchinstall.sh DESTDIR="$(pwd)" --backend=stg wined3d-DXTn
```

If you have any ideas for improvement, feel free to open a new bug report for that or to write patches. Please note that patchinstall.sh is autogenerated from debian/tools/patchinstall.sh.in.
